modules/data.atmosphere/R/downscale.met.R

```diff
@@ -0,0 +1,420 @@
+      # of the upscaled flux value within the window, and step through that window sequentially so that we are only using
+      # the standard devaition of values recorded at the same time as the value we are generating.
+      randtemp <- vector()
+      for (x in seq_along(model$air_temperature)){
+        four <- vector()
+        for (n in seq_along(1:div)){
+          lowday <- x*div-wd*sd_step+n
+          highday <- x*div+wd*sd_step+n
+          if (lowday < 0){lowday = x+n
+          highday = x+n+wd*sd_step+n}
+          if (highday > reso_len){
+            highday = reso_len
+            lowday = x*div-wd*sd_step+n}
+          four[n] <- rnorm(1, mean = model$air_temperature[x],
+                            sd = sd(upscale_flux$air_temperature[seq(from=lowday,to=highday,by=sd_step)]))}
+        randtemp <- append(randtemp,four)
+      }
+      
+      # Air Temperature Max and Min
+      temp_max <- vector()
+      if (all(is.na(model$air_temperature_max)) == TRUE){
+        temp_max <- rep(NA,reso_len)}
+      if (all(is.na(model$air_temperature_max)) == FALSE){
+        for (x in seq_along(model$air_temperature_max)){
+          four <- vector()
+          for (n in seq_along(1:div)){
+            lowday <- x*div-wd*sd_step+n
+            highday <- x*div+wd*sd_step+n
+            if (lowday < 0){lowday = x+n
+            highday = x+n+wd*sd_step+n}
+            if (highday > reso_len){
+              highday = reso_len
+              lowday = x*div-wd*sd_step+n}
+            four[n] <- rnorm(1, mean = model$air_temperature_max[x],
+                             sd = sd(upscale_flux$air_temperature_max[seq(from=lowday,to=highday,by=sd_step)]))}
+        temp_max <- append(temp_max,four)}
+      }
+      
+      temp_min <- vector()
+      if (all(is.na(model$air_temperature_min)) == TRUE){
+        temp_min <- rep(NA,reso_len)}
+      if (all(is.na(model$air_temperature_min)) == FALSE){
+        for (x in seq_along(model$air_temperature_min)){
+          four <- vector()
+          for (n in seq_along(1:div)){
+            lowday <- x*div-wd*sd_step+n
+            highday <- x*div+wd*sd_step+n
+            if (lowday < 0){lowday = x+n
+            highday = x+n+wd*sd_step+n}
+            if (highday > reso_len){
+              highday = reso_len
+              lowday = x*div-wd*sd_step+n}
+            four[n] <- rnorm(1, mean = model$air_temperature_min[x],
+                             sd = sd(upscale_flux$air_temperature_min[seq(from=lowday,to=highday,by=sd_step)]))}
+          temp_min <- append(temp_min,four)}
+      }
+      
+      # Precipitation_flux
+      # this takes the daily total of precipitation and uses that as a total possible amount of precip. 
+      # It randomly distributes the values of precipitation
+      rand_vect_cont <- function(N, M, sd = 1) {
+        vec <- rnorm(N, M/N, sd)
+        vec / sum(vec) * M
+      }
+      precip <- vector()
+      for (x in seq_along(model$precipitation_flux)){
```

Comment:
@araiho thanks for the comments and pointing me to formatR, this helps a lot. Will make changes and hopefully push to master before telecon tomorrow